### PR TITLE
Make sure global variable is handled properly

### PR DESCRIPTION
**Pull request by hkaiser**
Make sure global variable is

a) initialized on demand only
b) freed during shutdown

### CODE REVIEW

File: libs/lcos_local/src/composable_guard.cpp [hunk 1]
```diff
@@ -225,12 +225,36 @@ namespace hpx { namespace lcos { namespace local {
...
+            empty_helper() = default;
+            ~empty_helper()
+            {
+                auto& empty = get_empty_guard_task();
+                delete empty;
+                empty = nullptr;
+            }
+        };
+
+        empty_helper empty_helper_{};
```
Comment (aurianer) [inline libs/lcos_local/src/composable_guard.cpp:246]: Why is this needed ? (still working on my c++ skills) ^^'

Comment (hkaiser) [reply libs/lcos_local/src/composable_guard.cpp:246]: This creates a global instance of `empty_helper`. We rely on the destructor of this (global) object to clean up the allocated memory.

**Comment by stevenrbrandt** [comment]
Sorry, for whatever reason, I'm just seeing this ticket now.
